### PR TITLE
Adding Informer config to Meshsync CRD

### LINKED ISSUE #479: Meshery Operator to re-configure Meshsync Informer
#### Current Behavior
Meshsync informer watches every component (ConfigMap, Pods, etc).

#### Desired Behavior
Meshsync informer should be configurable by the user.

#### Implementation
Phase 1: 
- the configuration will be a simple blacklist of components to not watch

Phase 2:
- the configuration will be a separate object that will include the components to watch, a timeout per component (TBD).

#### Acceptance Tests
User should be able to configure which components to watch and how frequently to receive updates from Meshsync regarding them.

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Azanul**
**Description**

This PR fixes #479 

**Notes for Reviewers**
https://github.com/meshery/meshsync/pull/199
https://github.com/meshery/meshery/pull/7721

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
What tests do we have?

Is there a docs PR associated with this?

**Comment by Aisuko** [comment]
Hi guys, I am not familiar with the contents. Please call more people here to have a review. Thanks.

File: controllers/meshsync_controller.go [hunk 1]
```diff
@@ -153,7 +153,7 @@ func (r *MeshSyncReconciler) reconcileMeshsync(ctx context.Context, enable bool,
 		if enable {
 			er := r.Update(ctx, object)
 			if er != nil {
-				return ctrl.Result{}, ErrUpdateMeshsync(er)
+				return ctrl.Result{}, ErrUpdateResource(er)
```
Comment (leecalcote) [inline controllers/meshsync_controller.go:156]: Are we missing the complete error code definition?

Comment (leecalcote) [reply controllers/meshsync_controller.go:156]: Probable cause and suggested remediation?

Comment (Azanul) [reply controllers/meshsync_controller.go:156]: Open for suggestions

**Comment by Aisuko** [comment]
Hi, @Chadha93 why close this PR? We need more detail for a reason close PR. I believe @Azanul is working on this.

**Comment by Chadha93** [comment]
> Hi, @Chadha93 why close this PR? We need more detail for a reason close PR. I believe @Azanul is working on this.

Hi @Aisuko, Azan has moved on. He is no longer contributing. We are closing out all his PRs and not accepting any of his work as his departure is not on good terms, by not abiding by the community guidelines. He has been removed from all community places and GitHub orgs.

As I close out this PR, and we need to reassign this work to another contributor.

**Comment by Aisuko** [comment]
Alright, as I discussed with Lee we should close this PR.

**Pull request closed without merging**